File: Loenn/entities/wishAltar.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local wishAltar = {}

wishAltar.name = "MaggyHelper/WishAltar"
wishAltar.depth = -100
wishAltar.fieldInformation = {
    requiredHearts = {
        minimumValue = 0
    }
}
wishAltar.placements = {
    {
        name = "normal",
        data = {
            canInteract = true,
            dialoguePrefix = "WISH_ALTAR",
            requiredHearts = 0
        }
    }
}

function wishAltar.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("objects/wish_altar/inactive", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    sprite.color = {0.8, 0.6, 1.0, 1.0}
    return sprite
end

function wishAltar.selection(room, entity)
    return utils.rectangle(entity.x - 24, entity.y - 48, 48, 48)
end

return wishAltar

File: framework/extracted/Ahorn/entities/WishAltar.jl
module WishAltar

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/WishAltar" WishAltar(x::Integer, y::Integer, canInteract::Bool=true, dialoguePrefix::String="WISH_ALTAR", requiredHearts::Integer=0)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(WishAltar)
)

function Ahorn.selection(entity::WishAltar)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::WishAltar, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end
